lib/lambda_ethereum_consensus/state_transition/mutators.ex

```diff
@@ -160,4 +160,61 @@ defmodule LambdaEthereumConsensus.StateTransition.Mutators do
   defp whistleblower_index(whistleblower_index, proposer_index) do
     if whistleblower_index == nil, do: proposer_index, else: whistleblower_index
   end
+
+  @spec apply_deposit(
+          BeaconState.t(),
+          SszTypes.bls_pubkey(),
+          SszTypes.bytes32(),
+          SszTypes.uint64(),
+          SszTypes.bls_signature()
+        ) :: {:ok, BeaconState.t()} | {:error, binary()}
+  def apply_deposit(state, pubkey, withdrawal_credentials, amount, signature) do
+    validator_pubkeys = Enum.map(state.validators, fn validator -> validator.pubkey end)
+
+    if Enum.member?(validator_pubkeys, pubkey) do
+      index = Enum.find_index(validator_pubkeys, fn validator -> validator == pubkey end)
+      state = increase_balance(state, index, amount)
+      {:ok, state}
+    else
+      deposit_message = %SszTypes.DepositMessage{
+        pubkey: pubkey,
+        withdrawal_credentials: withdrawal_credentials,
+        amount: amount
+      }
+
+      domain = Misc.compute_domain(Constants.domain_deposit())
+
+      signing_root = Misc.compute_signing_root(deposit_message, domain)
+
+      case Bls.verify(pubkey, signing_root, signature) do
+        {:ok, verified} ->
+          apply_initial_deposit(verified, state, pubkey, withdrawal_credentials, amount)
+
+        {:error, msg} ->
+          {:error, msg}
+      end
+    end
+  end
+
+  defp apply_initial_deposit(verified, state, pubkey, withdrawal_credentials, amount) do
+    if verified do
```

Comment:
Why do we need to check if verified is a truthy value? If that's the case we can include it in the `with` as:

```elixir
with {:ok, true} <- Bls.verify... do
  %Beaconstate{...}
else
  {:ok, _} -> {:error, "BLS verification failed"}
  other -> other
end
```

Reply:
I am checking if verified is equal to true actually